coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala

```diff
@@ -63,6 +64,12 @@ class ShardKeyRegexPlanner(dataset: Dataset,
     LogicalPlan.getRawSeriesFilters(logicalPlan)
       .map { s => s.filter(f => dataset.options.nonMetricShardColumns.contains(f.column))}
 
+  /**
+    * Returns true when shard key filters do not have regex
+    */
+  private def withoutShardKeyRegex(logicalPlan: LogicalPlan) = getNonMetricShardKeyFilters(logicalPlan).
+    forall(_.forall(f => !f.filter.isInstanceOf[EqualsRegex] && !f.filter.isInstanceOf[NotEqualsRegex]))
```

Comment:
It would be better if you can just check for `Equals` instead of doing other way. Just a suggestion.